base/lib/src/storage/task/request_task.dart

```diff
@@ -43,30 +46,109 @@ abstract class RequestTask<T> extends Task<T> {
 
   /// [createTask] 被取消后触发
   @mustCallSuper
-  void postCancel(DioError error) {
+  void postCancel(StorageError error) {
     controller?.notifyStatusListeners(RequestTaskStatus.Cancel);
   }
 
   @override
   @mustCallSuper
-  void postError(Object error) {
-    // 通知状态
-    if (error is DioError && error.type == DioErrorType.CANCEL) {
-      postCancel(error);
-    } else {
-      controller?.notifyStatusListeners(RequestTaskStatus.Error);
+  void postError(Object error) async {
+    // 重试和冻结
+    if (error is DioError) {
+      if (!canConnectToHost(error)) {
+        // host 连不上，判断是否 host 不可用造成的, 比如 tls error(没做还)
+        if (isHostUnavailable(error)) {
+          config.hostProvider.freezeHost(error.request.path);
+        }
+
+        // 继续尝试当前 host，如果是服务器坏了则切换到其他 host
+        if (_retryCount < config.retryLimit) {
+          _retryCount++;
+          manager.restartTask(this);
+          return;
+        }
+      }
+
+      // 能连上但是服务器不可用，比如 502
+      if (isHostUnavailable(error)) {
+        config.hostProvider.freezeHost(error.request.path);
```

Comment:
> 冻结 `path`?

确定一下，因为 `host`、`path` 从定义上是不同的东西

Reply:
`error.request.path as Host` 你就能理解了把？
这不是我们的业务代码没有区分 `host` 和 `path`

Reply:
我的意思是：
这里 `dio` 说是 `path`，可能真的会拿到 `path`，而不是 `host`

Reply:
dio 这边 path 不太一样，可以 https 开头

> If the `path` starts with 'http(s)', the `baseURL` will be ignored, otherwise, it will be combined and then resolved with the baseUrl.

Reply:
你这个引用看起来像是某个参数的，不像是这个返回值的

Reply:
就是 path 的注释。不用我翻译了把

Reply:
你也说是可以，不是只有
也就是同时支持 `HTTPURL` 和 `path` 嘛，那我给个 `path` 你能正常冻结吗
你现在是因为碰巧没有使用 `baseUrl` 的方式去使用才没问题而已

Reply:
dio 的意思是如果 http(s) 开头就是全名比如(https://up.qiniu.com)。
其他情况会拼接上 baseURL，比如 baseURL 是 `https://up.qiniu.com`, 代码是 `post('/abc')`, path 就是 `https://up.qiniu.com/abc`

Reply:
> 其他情况会拼接上 baseURL，比如 baseURL 是 `https://up.qiniu.com`, 代码是 `post('/abc')`, path 就是 `https://up.qiniu.com/abc`

`path` 是 `/abc`，看实现去

Reply:
先不纠结，后面再确认